docs/apis-tools/spring-zeebe-sdk/configuration.md

```diff
@@ -55,507 +70,588 @@ This applies the following defaults:
 https://github.com/camunda/camunda/blob/main/clients/spring-boot-starter-camunda-sdk/src/main/resources/modes/self-managed.yaml
 ```
 
-### Basic
+## Connectivity
 
-This allows you to connect to a Self-Managed instance protected with basic authentication. The default URLs are configured to align with all Camunda distributions using `localhost` addresses.
+To connection to Camunda is determined by `camunda.client.grpc-address` and `camunda.client.rest-address`
 
-Activate by setting:
+### Camunda API connection
+
+#### gRPC address
+
+Define the address of the [gRPC API](/apis-tools/zeebe-api/grpc.md) exposed by the [Zeebe Gateway](/self-managed/zeebe-deployment/zeebe-gateway/zeebe-gateway-overview.md):
 
 ```yaml
 camunda:
   client:
-    mode: basic
+    grpc-address: http://localhost:26500
 ```
 
-This applies the following defaults:
+:::note
+You must add the `http://` scheme to the URL to avoid a `java.lang.NullPointerException: target` error.
+:::
 
-```yaml reference referenceLinkText="Source" title="Basic mode"
-https://github.com/camunda/camunda/blob/main/clients/spring-boot-starter-camunda-sdk/src/main/resources/modes/basic.yaml
+#### REST address
+
+Define address of the [Camunda 8 REST API](/apis-tools/camunda-api-rest/camunda-api-rest-overview.md) exposed by the [Zeebe Gateway](/self-managed/zeebe-deployment/zeebe-gateway/zeebe-gateway-overview.md):
+
+```yaml
+camunda:
+  client:
+    rest-address: http://localhost:8080
 ```
 
-## Job worker configuration options
+:::note
+You must add the `http://` scheme to the URL to avoid a `java.lang.NullPointerException: target` error.
+:::
 
-### Job type
+#### REST over gRPC
 
-You can configure the job type via the `JobWorker` annotation:
+If true, the Camunda Client will use REST instead of gRPC whenever possible to communicate with the Camunda APIs:
 
-```java
-@JobWorker(type = "foo")
-public void handleJobFoo() {
-  // handles jobs of type 'foo'
-}
+```yaml
+camunda:
+  client:
+    prefer-rest-over-grpc: true
 ```
 
-If you don't specify the `type` attribute, the **method name** is used by default if you enabled the [`-parameters` compiler flag](/apis-tools/spring-zeebe-sdk/getting-started.md#enable-the-java-compiler--parameters-flag) in the [getting started section](/apis-tools/spring-zeebe-sdk/getting-started.md):
+### Advanced connectivity settings
 
-```java
-@JobWorker
-public void foo() {
-    // handles jobs of type 'foo'
-}
-```
+#### Keep alive
 
-As a third possibility, you can set a default job type:
+Time interval between keep alive messages sent to the gateway (default is 45s):
```

Comment:
❓ shall we outline when this is relevant?

Edit: I guess this is what you meant,. by refactoring the config docs to capture the actual user journey and use cases instead of just reference docs? Let's handle in a follow-up indeed.